Partitioned_Heat_Conduction/README.md

```diff
@@ -0,0 +1,39 @@
+developped in the adapter, it is a simple 2D problem, a 2x2 square plate
+where the boundaries temperature are set according to equation 1 + x^2 + 2 * y^2.
+
+# Example Files
+* README.md File          : Contains instructions for running example problem 
+* case.sif File           : This is the system input file generated by ElmerGUI(or written by hand), to be fed to ElmerSolver with the mesh files to run simulation 
+* Reference_Problem.geo   : This is the gmsh file to build gemoerty and generate the .msh file that defines mesh 
+* Reference_Problem.msh   : This is the msh file that mesh files for ElmerSolver can be generated from 
+
+# Generating Mesh Files
+It can be done in 2 ways
+* using ElmerGrid : just type ElmerGrid 14 2 Reference_Problem_grid.msh and the mesh files will be generated in a seperate folder under same name
+* using ElmerGUI  : open the msh file in ElmerGUI, define problem, save the project and generate .sif file, the mesh files will be generated in the location defined
+                  in the model/setup under MeshDB parameter. For more information please refer to ElmerGUI Tutorials 
+                  http://www.nic.funet.fi/pub/sci/physics/elmer/doc/ElmerTutorials.pdf
+
+# Generating .msh File
+These steps are required if the user want to regenerate the existing .msh file, gmsh software is required, open the .geo in gmsh by running command
+gmsh <FileName>.geo and follow the following tutorial https://www.youtube.com/watch?v=O1FyiBBuN98&ab_channel=JoshTheEngineer
+
+# Running Reference Example
+* Step 1    : Generate msh Files, type ElmerGrid 14 2 Reference_Problem.msh -out Reference_Problem_Mesh
+* Step 2    : Run Simulation, type ElmerSolver Reference_Problem.sif and wait for simulation to finish
+* Step 3    : visulaize results, results should exist in Results folder, run paraview and open .vtu files to se solution
+* Hint      : Results in other examples not necessarily exist in Results folder, the location can be modified in .sif file
+
+# Running Dirichlet Example
+* Step 1    : Generate msh Files, type ElmerGrid 14 2 Dirichlet_Participant.msh -out Dirichlet_Participant_Mesh
+* Step 2    : Run Simulation, type ElmerSolver Dirichlet_Participant.sif and wait for simulation to finish
+* Step 3    : visulaize results, results should exist in Results folder, run paraview and open .vtu files to se solution
+* Hint      : Results in other examples not necessarily exist in Results folder, the location can be modified in .sif file
+
+# Running Neumann Example
+* Step 1    : Generate msh Files, type ElmerGrid 14 2 Neumann_Participant.msh -out Neumann_Participant_Mesh
+* Step 2    : Run Simulation, type ElmerSolver Dirichlet_Participant.sif and wait for simulation to finish
+* Step 3    : visulaize results, results should exist in Results folder, run paraview and open .vtu files to se solution
+* Hint      : Results in other examples not necessarily exist in Results folder, the location can be modified in .sif file
```

Comment:
I tried to run the Elmer-Elmer version of this case. I think the explanation here is generally useful. Please format the code using `code_formatting` in markdown to make the workflow more clear. I would also suggest to explain the compilation step here again, since otherwise it is not 100% clear what to do and from where the command has to be executed (I'm not sure whether I actually did it right below).

### Compile the adapter

As described in the `elmer-adapter/README.md`: 

```
elmer-adapter/Partitioned_Heat_Conduction$ elmerf90 -o CouplerSolver.so Coupler_Solver.F90 -lprecice
```

### Prepare the case

The following steps are executed for both solvers (`Dirichlet_Participant` and `Neumann_Participant`). I will only describe `Dirichlet_Participant` here.

```
elmer-adapter/Partitioned_Heat_Conduction/Dirichlet_Participant$ ElmerGrid 14 2 Dirichlet_Participant.msh -out Dirichlet_Participant_Mesh
```

### Run the case

Has to be done for both participants at the same time:

```
elmer-adapter/Partitioned_Heat_Conduction/Dirichlet_Participant$ ElmerSolver case.sif 
```

### Results

* I was able to run explicit coupling and there were `.vtu` files in the `out` folder.
* Implicit coupling did not complete, but there are also some `.vtu` files in the `out` folder.
* There is reasonable output written to the console by preCICE. It looks like the two participants are able to build up a connection and a coupling is established.
* Looking at the files in paraview, I did not see reasonable output. But this might also be wrong usage on my side.

Reply:
Hi i checked your steps and the code as well, you are correct, but some modification, is needed, i modified the read me as well to make more suitable for running the simulation, you may check it and try it without looking at my comments to run the simulation from a user perspective. i explained a little bit more about the compilation in the read me files; the general one in the main directory as well as the readme file in the partitioned heat plate problem.

# Compile the adapter
`elmerf90 -o Coupler_Solver.so Coupler_Solver.F90 /usr/lib/x86_64-linux-gnu/libprecice.so.2`
you have to link preCICE by providing the path of the shared library, for locating the path check the readme

Second step does not need modification
# Prepare the case
`elmer-adapter/Partitioned_Heat_Conduction/Dirichlet_Participant$ ElmerGrid 14 2 Dirichlet_Participant.msh -out Dirichlet_Participant_Mesh`

# Run the case
Has to be done for both participants at the same time in 2 different terminals:
`elmer-adapter/Partitioned_Heat_Conduction/Dirichlet_Participant$ ElmerSolver case.sif`

# Results
* The results are located as .vtu files in the out folder.
* only explicit coupling is allowed since no access to the time loop of Elmer
* Console should print info where communication is established, results from adapter should be printed in red, data printed from adapter just developing purpose, printing could be removed from adapter code
* Check the files at the paraview, this time results should be correct, the reason why you had wrong results is that the FEniCS solver has a different direction of the normal flux, so the data has to be inverted(multiplied by -1) before copying it to Elmer, this is done in the adapter code, i removed the multiplication and set the adapter for Elmer-Elmer case, and pushed it, so the current code on the repository works for Elmer-Elmer case. If you want to run the Elmer-FEniCS case the adapter code must be modified again to invert the flux

Reply:
Hi i checked your steps and the code as well, you are correct, but some modification, is needed, i modified the read me as well to make more suitable for running the simulation, you may check it and try it without looking at my comments to run the simulation from a user perspective. i explained a little bit more about the compilation in the read me files; the general one in the main directory as well as the readme file in the partitioned heat plate problem.

# Compile the adapter
`elmerf90 -o Coupler_Solver.so Coupler_Solver.F90 /usr/lib/x86_64-linux-gnu/libprecice.so.2`
you have to link preCICE by providing the path of the shared library, for locating the path check the readme

Second step does not need modification
# Prepare the case
`elmer-adapter/Partitioned_Heat_Conduction/Dirichlet_Participant$ ElmerGrid 14 2 Dirichlet_Participant.msh -out Dirichlet_Participant_Mesh`

# Run the case
Has to be done for both participants at the same time in 2 different terminals:
`elmer-adapter/Partitioned_Heat_Conduction/Dirichlet_Participant$ ElmerSolver case.sif`

# Results
* The results are located as .vtu files in the out folder.
* only explicit coupling is allowed since no access to the time loop of Elmer
* Console should print info where communication is established, results from adapter should be printed in red, data printed from adapter just developing purpose, printing could be removed from adapter code
* Check the files at the paraview, this time results should be correct, the reason why you had wrong results is that the FEniCS solver has a different direction of the normal flux, so the data has to be inverted(multiplied by -1) before copying it to Elmer, this is done in the adapter code, i removed the multiplication and set the adapter for Elmer-Elmer case, and pushed it, so the current code on the repository works for Elmer-Elmer case. If you want to run the Elmer-FEniCS case the adapter code must be modified again to invert the flux